### PR TITLE
test: [GroupLibraryMangaUseCase] test coverage

### PR DESCRIPTION
**Pull request by nonproto**
💡 What: Added unit test suite `GroupLibraryMangaUseCaseTest.kt` for `GroupLibraryMangaUseCase` to increase test coverage.
🎯 Why: Covers the edge cases for grouping library items by dynamic states (Language, Track Status), ungrouped state, and categorized states to protect against regressions.

---
*PR created automatically by Jules for task [7929323682914439519](https://jules.google.com/task/7929323682914439519) started by @nonproto*

### CODE REVIEW

File: app/src/test/java/org/nekomanga/usecases/library/GroupLibraryMangaUseCaseTest.kt [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        // Assert
+        assertEquals(2, result.size)
+
+        // Items are sorted by key based on the String.CASE_INSENSITIVE_ORDER comparator of
+        // ByLanguage
+        val enCategory = result.first { it.categoryItem.name == "en" }
+        assertEquals(2, enCategory.libraryItems.size)
+
+        val jaCategory = result.first { it.categoryItem.name == "ja" }
+        assertEquals(1, jaCategory.libraryItems.size)
```
[the remark this answers is not in the document]
Comment (nonproto) [reply app/src/test/java/org/nekomanga/usecases/library/GroupLibraryMangaUseCaseTest.kt:72]: > medium
> 
> The test verifies that categories are created but doesn't check the order of the results or the properties of the generated `CategoryItem` objects (like `id` and `isDynamic`). Since `groupByDynamic` is responsible for both grouping and sorting, and it generates dynamic IDs based on the sorted order, these aspects should be asserted. Additionally, consider testing a manga with multiple languages to verify it correctly appears in multiple groups. Use `assertTrue` for boolean assertions as per repository guidelines.
> 
> References

@jules

File: app/src/test/java/org/nekomanga/usecases/library/GroupLibraryMangaUseCaseTest.kt [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    fun `groupByDynamic groups items correctly by track status including fallbacks`() {
+        // Arrange
+        val trackedManga = mockLibraryMangaItem(mangaId = 1L)
+        val untrackedManga = mockLibraryMangaItem(mangaId = 2L)
+
+        val mangaList = listOf(trackedManga, untrackedManga)
+        val currentGroup = LibraryGroup.ByTrackStatus
+        val sortOrder = LibrarySort.Title
+        val sortAscending = true
+        val trackStatusMap = mapOf(1L to listOf("Reading"))
```
[the remark this answers is not in the document]
Comment (nonproto) [reply app/src/test/java/org/nekomanga/usecases/library/GroupLibraryMangaUseCaseTest.kt:85]: > medium
> 
> The current test uses "Reading" and "Not tracked", which both fall into the default case (weight 0) of the `ByTrackStatus` comparator. To properly verify the sorting logic, it's recommended to use statuses with different weights defined in `LibraryGroup.ByTrackStatus` (e.g., "Completed" vs "Dropped").

@jules

File: app/src/test/java/org/nekomanga/usecases/library/GroupLibraryMangaUseCaseTest.kt [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+        // Assert
+        assertEquals(1, result.size)
+
+        val categoryGroup = result.first()
+        assertEquals(CategoryItem.ALL_CATEGORY, categoryGroup.categoryItem.name)
+        // Should distinct by mangaId, so 2 items total
+        assertEquals(2, categoryGroup.libraryItems.size)
+        assertEquals(1L, categoryGroup.libraryItems[0].displayManga.mangaId)
+        assertEquals(2L, categoryGroup.libraryItems[1].displayManga.mangaId)
```
[the remark this answers is not in the document]
Comment (nonproto) [reply app/src/test/java/org/nekomanga/usecases/library/GroupLibraryMangaUseCaseTest.kt:139]: > medium
> 
> The test should verify the specific properties of the `CategoryItem` created by `groupByUngrouped`, specifically that the `id` is `-1` and `isDynamic` is `true`, as these are hardcoded requirements in the use case. Use `assertTrue` for boolean assertions.
> 
> ```kotlin
>         assertEquals(1, result.size)
> 
>         val categoryGroup = result.first()
>         assertEquals(CategoryItem.ALL_CATEGORY, categoryGroup.categoryItem.name)
>         assertEquals(-1, categoryGroup.categoryItem.id)
>         assertTrue(categoryGroup.categoryItem.isDynamic)
>         // Should distinct by mangaId, so 2 items total
>         assertEquals(2, categoryGroup.libraryItems.size)
>         assertEquals(1L, categoryGroup.libraryItems[0].displayManga.mangaId)
>         assertEquals(2L, categoryGroup.libraryItems[1].displayManga.mangaId)
> ```
> 
> References

@jules

File: app/src/test/java/org/nekomanga/usecases/library/GroupLibraryMangaUseCaseTest.kt [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+            CategoryItem(id = 0, name = CategoryItem.SYSTEM_CATEGORY, sortOrder = LibrarySort.Title)
+
+        val categories = listOf(category100, category200, systemCategoryEmpty)
+
+        // Act
+        val result =
+            useCase.groupByCategory(libraryMangaList = mangaList, categoryList = categories)
+
+        // Assert
+        assertEquals(2, result.size)
```
[the remark this answers is not in the document]
Comment (nonproto) [reply app/src/test/java/org/nekomanga/usecases/library/GroupLibraryMangaUseCaseTest.kt:165]: > medium
> 
> The test verifies that empty system categories are filtered out, but it doesn't verify that empty _non-system_ categories are preserved. Adding a case for an empty regular category would ensure the conditional logic in `groupByCategory` is fully covered and doesn't regress to filtering all empty categories.

@jules
